rust/otap-dataflow/crates/otap/src/syslog_cef_receiver.rs

```diff
@@ -245,28 +359,48 @@ impl local::Receiver<OtapPdata> for SyslogCefReceiver {
                                                 _ = interval.tick() => {
                                                     if arrow_records_builder.len() > 0 {
                                                         // Build the Arrow records and send them
-                                                        let arrow_records = arrow_records_builder.build().expect("Failed to build Arrow records");
+                                                        let items = u64::from(arrow_records_builder.len());
+                                                        let arrow_records = arrow_records_builder
+                                                            .build()
+                                                            .expect("Failed to build Arrow records");
 
                                                         // Reset the builder for the next batch
                                                         arrow_records_builder = ArrowRecordsBuilder::new();
 
-                                                        if let Err(_e) = effect_handler.send_message(OtapPdata::new_todo_context(arrow_records.into())).await {
-                                                            return; // Break out of the entire task
+                                                        match effect_handler
+                                                            .send_message(OtapPdata::new_todo_context(arrow_records.into()))
+                                                            .await
+                                                        {
+                                                            Ok(_) => {
+                                                                let mut m = metrics.borrow_mut();
+                                                                m.received_logs_success.add(items);
+                                                            }
+                                                            Err(_) => {
+                                                                let mut m = metrics.borrow_mut();
+                                                                m.received_logs_refused.add(items);
+                                                                m.tcp_connections_active.dec();
```

Comment:
The TCP connection would still be active. It's closed only on EOF or socket read error.

```suggestion
```